Sources/CoffeeAddictsLibrary/Commands/CoffeeShopsFinder.swift

```diff
@@ -0,0 +1,62 @@
+//
+//  File.swift
+//  
+//
+//  Created by Mihai Garda Popescu on 23.07.2021.
+//
+
+import ArgumentParser
+import CoreLocation
+
+public enum Command {}
+
+extension Command {
+    public struct CoffeeShopsFinder: ParsableCommand {
+        public init() {}
+        public static var configuration = CommandConfiguration(
+            commandName: "find-shops",
+            abstract: "Returns a list of the three closest coffee shops"
+        )
+
+        @Argument(help: "Your X Coordinate")
+        var xCoord: Double
+
+        @Argument(help: "Your Y Coordinate")
+        var yCoord: Double
+
+        @Argument(help: "Coffee Shops CSV file URL")
+        var csvUrl: String
+
+        public func run() throws {
+            CoffeeShopsCSVDownloader.downloadCSVFileFromURLString(csvUrl) { csvString, status in
+                if status == .CSVFileDownloadSuccess {
+                    if let string = csvString {
```

Comment:
Having multiple nested if statements looks weird. It's there a way to try to break the logic and make the code more readable?

Reply:
Done, used a guard and moved part of the logic into a separate method.